lib-sql/tiger_import_start.sql

```diff
@@ -71,18 +117,34 @@ BEGIN
   place_centroid := ST_Centroid(linegeo);
   out_partition := get_partition('us');
 
-  out_parent_place_id := getNearestNamedRoadPlaceId(out_partition, place_centroid,
-                                                    token_info);
+  -- HYBRID LOOKUP LOGIC (see gh-issue #2463)
+  -- if partition tables exist, use them for fast spatial lookups
+  {% if 'location_road_0' in db.tables %}
+    out_parent_place_id := getNearestNamedRoadPlaceId(
+      out_partition, place_centroid, token_info
+    );
 
-  IF out_parent_place_id IS NULL THEN
-    SELECT getNearestParallelRoadFeature(out_partition, linegeo)
-      INTO out_parent_place_id;
-  END IF;
+    IF out_parent_place_id IS NULL THEN
+      SELECT getNearestParallelRoadFeature(out_partition, linegeo)
+        INTO out_parent_place_id;
+    END IF;
 
-  IF out_parent_place_id IS NULL THEN
-    SELECT getNearestRoadPlaceId(out_partition, place_centroid)
-      INTO out_parent_place_id;
-  END IF;
+    IF out_parent_place_id IS NULL THEN
+      SELECT getNearestRoadPlaceId(out_partition, place_centroid)
+        INTO out_parent_place_id;
+    END IF;
+
+  -- When updatable information has been dropped:
+  -- Partition tables no longer exist, but search_name still persists.
+  {% elif 'search_name' in db.tables %}
+    RAISE WARNING 'Database frozen. Performing fallback tiger import lookup without partition tables';
+    -- Fallback: Look up in 'search_name' table 
+    -- though spatial lookups here can be slower.
+    out_parent_place_id := lookup_road_in_search_name(place_centroid, token_info);
+  {% else %}
```

Comment:
This else will never be reached (the Python code already checks for the condition), so I would suggest to just leave it out completely.